add-schema.sh

```diff
@@ -38,69 +38,102 @@ fi
+\    </profile>
 SED_SCRIPT
 fi
 
 
 # Add schema dependency in web/pom.xml
 line=$(grep -n "schema-${schema}" web/pom.xml | cut -d: -f1)
 
-if [ ! $line ]
+if [ ! -n "$line" ]
 then
-  line=$(grep -n 'schema-iso19139</artifactId>' web/pom.xml | cut -d: -f1)
-  insertLine=$(($line + 2))
+  gnSchemasVersion='${project.version}'
+  basedir='${basedir}'
 
-  projectGroupId='${project.groupId}'
-  gnSchemasVersion='${gn.schemas.version}'
-
-  echo "Adding schema ${schema} dependency to web/pom.xml"
+  echo "Adding schema ${schema} dependency to web/pom.xml for schemaCopy"
 
+  line=$(grep -n '<!-- add schema_plugins -->' web/pom.xml | cut -d: -f1 | tail -1)
+  insertLine=$(($line))
   sed $sedopt -f /dev/stdin web/pom.xml << SED_SCRIPT
   ${insertLine} a\\
-\    <dependency>\\
-\      <groupId>${projectGroupId}</groupId>\\
-\      <artifactId>schema-${schema}</artifactId>\\
-\      <version>${gnSchemasVersion}</version>\\
-\    </dependency>
+\        <dependency>\\
+\          <groupId>org.geonetwork-opensource.schemas</groupId>\\
+\          <artifactId>schema-${schema}</artifactId>\\
+\          <version>${gnSchemasVersion}</version>\\
+\        </dependency>
 SED_SCRIPT
-fi
 
+  echo "Adding schema ${schema} dependency to web/pom.xml for schemaUnpack"
 
-# Add schema resources in web/pom.xml
-line=$(grep -n "schemas/${schema}/src/main/plugin</directory>" web/pom.xml | cut -d: -f1)
-
-if [ ! $line ]
-then
-  line=$(grep -n 'schemas/iso19139/src/main/plugin</directory>' web/pom.xml | cut -d: -f1)
-  finalLine=$(($line + 3))
-
-  projectBaseDir='${project.basedir}'
-  baseDir='${basedir}'
-
-  echo "Adding schema ${schema} resources to web/pom.xml"
-
+  line=$(grep -n '</profiles>' web/pom.xml | cut -d: -f1)
+  insertLine=$(($line - 1))
   sed $sedopt -f /dev/stdin web/pom.xml << SED_SCRIPT
-  ${finalLine} a\\
-\                <resource>\\
-\                  <directory>${projectBaseDir}/../schemas/${schema}/src/main/plugin</directory>\\
-\                  <targetPath>${baseDir}/src/main/webapp/WEB-INF/data/config/schema_plugins</targetPath>\\
-\                </resource>
+  ${insertLine} a\\
+\    <profile>\\
+\      <id>schema-${schema}</id>\\
+\      <activation>\\
+\        <property><name>schemasCopy</name><value>!true</value></property>\\
+\        <file><exists>../schemas/${schema}</exists></file>\\
+\      </activation>\\
+\      <dependencies>\\
+\        <dependency>\\
+\          <groupId>org.geonetwork-opensource.schemas</groupId>\\
+\          <artifactId>schema-${schema}</artifactId>\\
+\          <version>${gnSchemasVersion}</version>\\
+\        </dependency>\\
+\      </dependencies>\\
+\      <build>\\
+\        <plugins>\\
+\          <plugin>\\
+\            <groupId>org.apache.maven.plugins</groupId>\\
+\            <artifactId>maven-dependency-plugin</artifactId>\\
+\            <executions>\\
+\              <execution>\\
+\                <id>${schema}-resources</id>\\
+\                <phase>process-resources</phase>\\
```

Comment:
This changes a bit the workflow used for developing schemas with GeoNetwork. Before:
1. You make changes to the schema in `schemas/$SCHEMA_FOLDER`.
2. Call `mvn process-resources` to copy the schema files to `web` folder

With this change:
1. You make changes to the schema in `schemas/$SCHEMA_FOLDER`.
2. Call `mvn install in `schemas/$SCHEMA_FOLDER`.
3. Call `mvn process-resources` to copy the schema files to `web` folder

Reply:
Understood, reverted.